databuilder/databuilder/task/search/search_metadata_to_elasticsearch_task.py

```diff
@@ -0,0 +1,140 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from typing import Dict, Iterator
+
+from pyhocon import ConfigTree
+
+from elasticsearch_dsl.connections import connections, Connections
+from elasticsearch.helpers import parallel_bulk
+from elasticsearch_dsl.document import Document
+from elasticsearch_dsl.index import Index
+
+from databuilder import Scoped
+from databuilder.task.base_task import Task
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.transformer.base_transformer import NoopTransformer, Transformer
+from databuilder.utils.closer import Closer
+
+from databuilder.task.search.document_mappings import RESOURCE_TO_MAPPING, english_analyzer, stemming_analyzer
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SearchMetadatatoElasticasearchTask(Task):
+    
+    
+    ELASTICSEARCH_CLIENT_CONFIG_KEY = 'client'
+    ENTITY_TYPE = 'doc_type'
+    MAPPING = 'document_mapping'
+    CUSTOM_ANALYZERS = 'custom_analyzers'
+    ELASTICSEARCH_NEW_INDEX_CONFIG_KEY = 'new_index'
+    ELASTICSEARCH_ALIAS_CONFIG_KEY = 'alias'
+
+    # config to control how many max documents to publish at a time
+    ELASTICSEARCH_PUBLISHER_BATCH_SIZE = 'batch_size'
+
+    def __init__(self,
+                 extractor: Extractor,
+                 transformer: Transformer = NoopTransformer()) -> None:
+        self.extractor = extractor
+        self.transformer = transformer
+
+        self._closer = Closer()
+        self._closer.register(self.extractor.close)
+        self._closer.register(self.transformer.close)
+
+    def init(self, conf: ConfigTree) -> None:
+        self.extractor.init(Scoped.get_scoped_conf(conf, self.extractor.get_scope()))
+        self.transformer.init(Scoped.get_scoped_conf(conf, self.transformer.get_scope()))
+
+        conf = Scoped.get_scoped_conf(conf, self.get_scope())
+        default_analyzers = [english_analyzer, stemming_analyzer]
+
+        self.entity = conf.get_string(SearchMetadatatoElasticasearchTask.ENTITY_TYPE, default='table').lower()
+        self.document_mapping = conf.get(SearchMetadatatoElasticasearchTask.MAPPING, RESOURCE_TO_MAPPING[self.entity])
+        # not sure if this is the best way to configure custom analyzers since it is not enough to add them to the documents
+        self.analyzers = conf.get_list(SearchMetadatatoElasticasearchTask.CUSTOM_ANALYZERS, default_analyzers)
+
+        self.elasticsearch_client = conf.get(SearchMetadatatoElasticasearchTask.ELASTICSEARCH_CLIENT_CONFIG_KEY)
+        self.elasticsearch_new_index = conf.get(SearchMetadatatoElasticasearchTask.ELASTICSEARCH_NEW_INDEX_CONFIG_KEY)
+        self.elasticsearch_alias = conf.get(SearchMetadatatoElasticasearchTask.ELASTICSEARCH_ALIAS_CONFIG_KEY)
+
+        self.elasticsearch_batch_size = conf.get(SearchMetadatatoElasticasearchTask.ELASTICSEARCH_PUBLISHER_BATCH_SIZE,
```

Comment:
nit: good to put defaults together in a defaults dict, like other databuilder components. Easier to see at a glance what has sensible defaults and what doesn't